include/boost/hana/fwd/concept/set_theoretic.hpp

```diff
@@ -0,0 +1,37 @@
+/*!
+@file
+Forward declares `boost::hana::SetTheoretic`.
+
+@copyright Shreyans Doshi 2017
+Distributed under the Boost Software License, Version 1.0.
+(See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+ */
+
+#ifndef BOOST_HANA_FWD_CONCEPT_SET_THEORETIC_HPP
+#define BOOST_HANA_FWD_CONCEPT_SET_THEORETIC_HPP
+
+#include <boost/hana/config.hpp>
+
+
+BOOST_HANA_NAMESPACE_BEGIN
+    //! @ingroup group-concepts
+    //! @defgroup group-SetTheoretic SetTheoretic
+    //! The `SetTheoretic` concept represents data structures supporting
+    //! set-theoretic functions like union, instersection, difference,
+    //! and symmetric_difference.
+    //!
+    //! Minimal complete definition
+    //! ---------------------------
+    //! `union_`, `intersection`, `difference` and `symmetric_difference`
```

Comment:
`symmetric_difference` can be derived from `union` and `difference`, see [this](https://en.wikipedia.org/wiki/Symmetric_difference#Properties). Hence, I don't think it should be part of the minimal complete definition.

Reply:
Makes sense. I think it should automatically kick in while calling difference first time if condition is not satisfied